### PR TITLE
Support to run xfstests with real disk

### PR DESCRIPTION
**Pull request by harish-24**
This patch allows to run xfstests on real disks

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: fs/xfstests.py [hunk 1]
```diff
@@ -83,16 +88,22 @@ def setUp(self):
...
+                           (line, os.path.join(self.srcdir, 'local.config')), shell=True)
+            line = ('export SCRATCH_DEV=%s' %
+                    self.scratch_dev).replace('/', '\/')
+            process.system('sed -i "s/export SCRATCH_DEV=.*/%s/g" %s' %
+                           (line, os.path.join(self.srcdir, 'local.config')), shell=True)
+        # mkfs for devices
+        if self.devices:
+            for dev in self.devices:
+                dev_obj = partition.Partition(dev)
+                dev_obj.mkfs(fstype=self.fs_to_test)
```
Comment (narasimhan-v) [inline fs/xfstests.py]: Are you not mounting this disk to `test_mnt` ?

File: fs/xfstests.py [hunk 2]
```diff
@@ -147,10 +158,10 @@ def tearDown(self):
         if os.path.exists(self.test_mnt):
             os.rmdir(self.test_mnt)
         if self.dev_type == 'loop':
-            for dev in self.loop_devices:
+            for dev in self.devices:
                 process.system('losetup -d %s' % dev, shell=True,
                                sudo=True, ignore_status=True)
-        self.part.unmount()
+            self.part.unmount()
```
Comment (narasimhan-v) [inline fs/xfstests.py]: unmount is needed for disks also, since we need to mount them after creating filesystem

File: fs/xfstests.py (hunk 1 above)
Comment (harish-24) [inline fs/xfstests.py]: The xfstests run will take care of mount to `test_mnt`

File: fs/xfstests.py (hunk 2 above)
Comment (harish-24) [inline fs/xfstests.py]: This unmount is for creating loop device ( '/mnt/loop-device' in this case). After loop devices are removed we are umounting it.

Comment (narasimhan-v) [inline fs/xfstests.py]: Ah okay, job.log would be helpful

**Comment by harish-24** [comment]
[xfstests_disk_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/860412/xfstests_disk_log.txt)

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks

**Pull request merged**